core/src/accounts_background_service.rs

```diff
@@ -18,25 +18,32 @@ const INTERVAL_MS: u64 = 100;
 const SHRUNKEN_ACCOUNT_PER_SEC: usize = 250;
 const SHRUNKEN_ACCOUNT_PER_INTERVAL: usize =
     SHRUNKEN_ACCOUNT_PER_SEC / (1000 / INTERVAL_MS as usize);
+const CLEAN_INTERVAL_SLOTS: u64 = 100;
 
 impl AccountsBackgroundService {
     pub fn new(bank_forks: Arc<RwLock<BankForks>>, exit: &Arc<AtomicBool>) -> Self {
         info!("AccountsBackgroundService active");
         let exit = exit.clone();
         let mut consumed_budget = 0;
+        let mut last_cleaned_slot = 0;
         let t_background = Builder::new()
             .name("solana-accounts-background".to_string())
             .spawn(move || loop {
                 if exit.load(Ordering::Relaxed) {
                     break;
                 }
-                let bank = bank_forks.read().unwrap().working_bank();
+                let bank = bank_forks.read().unwrap().root_bank().clone();
 
                 bank.process_dead_slots();
 
                 consumed_budget = bank
                     .process_stale_slot_with_budget(consumed_budget, SHRUNKEN_ACCOUNT_PER_INTERVAL);
 
+                if bank.block_height() - last_cleaned_slot > CLEAN_INTERVAL_SLOTS {
+                    bank.clean_accounts();
+                    last_cleaned_slot = bank.block_height();
```

Comment:
Yeah, This mostly randomizes the `clean_accounts()` timings across the cluster. But what about if we're doing coordinated restarts after crash? Every validators just idle on the same root thanks to the  `--wait-for-supemajority` and validators start at once. How about including some jitter like `thread_rng().gen_range(0, 10);` or something?

Reply:
ok; I can add in a follow-up. Shouldn't be that terrible even if all nodes are going at the same time considering most of the time is spent out of locks and shouldn't be using a ton of resources.